lib/pwwka/transmitter.rb

```diff
@@ -77,24 +79,40 @@ def self.send_message!(payload, routing_key,
             warn(resque_exception.message)
             raise e
           end
+
+        when :retry_async
+          begin
+            send_message_async(payload, routing_key, delay_by_ms: delayed ? delay_by || DEFAULT_DELAY_BY_MS : 0)
+          rescue => exception
+            warn(exception.message)
+            raise e
+          end
+
         else # ignore
       end
       false
     end
 
-    # Use Resque to enqueue the message.
+    # Enqueue the message with the configured background processor.
     # - :delay_by_ms:: Integer milliseconds to delay the message. Default is 0.
     def self.send_message_async(payload, routing_key,
                                 delay_by_ms: 0,
                                 type: nil,
                                 message_id: :auto_generate,
                                 headers: nil)
+      background_job_processor = Pwwka.configuration.background_job_processor
       job = Pwwka.configuration.async_job_klass
-      # Be perhaps too carefully making sure we queue jobs in the legacy way
-      if type == nil && message_id == :auto_generate && headers == nil
-        Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key)
-      else
-        Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key, type: type, message_id: message_id, headers: headers)
+
+      if background_job_processor == :resque
+        # Be perhaps too carefully making sure we queue jobs in the legacy way
+        if type == nil && message_id == :auto_generate && headers == nil
+          Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key)
+        else
+          Resque.enqueue_in(delay_by_ms/1000, job, payload, routing_key, type: type, message_id: message_id, headers: headers)
+        end
+      elsif background_job_processor == :sidekiq
+        options = { delay_by_ms: delay_by_ms, type: type, message_id: message_id, headers: headers }
+        job.perform_async(payload, routing_key, options)
       end
```

Comment:
i think this is perfectly fine.  another option though to clean it up could be to add an `enqueue` method to the `background_job_processor` that knows how to enqueue it's own type of job